src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -6,295 +6,193 @@
-        {
-          graphics.drawImage(movementImage, (int) (points[numTerritories - 1].x + textXOffset - xOffset),
-              (int) (points[numTerritories - 1].y + textyOffset - yOffset + translateY), null);
-          graphics.drawImage(movementImage, (int) (points[numTerritories - 1].x + textXOffset - xOffset),
-              (int) (points[numTerritories - 1].y + textyOffset - yOffset - translateY), null);
-        }
-        if (scrollWrapX && scrollWrapY) {
-          graphics.drawImage(movementImage, (int) (points[numTerritories - 1].x + textXOffset - xOffset + translateX),
-              (int) (points[numTerritories - 1].y + textyOffset - yOffset + translateY), null);
-          graphics.drawImage(movementImage, (int) (points[numTerritories - 1].x + textXOffset - xOffset - translateX),
-              (int) (points[numTerritories - 1].y + textyOffset - yOffset - translateY), null);
-        }
-        final Image cursorImage = routeDescription.getCursorImage();
-        if (cursorImage != null) {
-          graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset),
-              (int) (points[numTerritories - 1].y + cursorYOffset - yOffset), null);
-          if (scrollWrapX /* && !scrollWrapY */) {
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset + translateX),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset), null);
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset - translateX),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset), null);
-          }
-          if (/* !scrollWrapX && */scrollWrapY) {
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset + translateY), null);
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset - translateY), null);
-          }
-          if (scrollWrapX && scrollWrapY) {
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset + translateX),
-                (int) (points[numTerritories - 1].y + cursorXOffset - yOffset + translateY), null);
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset - translateX),
-                (int) (points[numTerritories - 1].y + cursorXOffset - yOffset - translateY), null);
-          }
-        }
-      }
-    } finally {
-      graphics.setTransform(original);
+    if (routeDescription == null) {
+      return;
+    }
+    final Route route = routeDescription.getRoute();
+    if (route == null) {
+      return;
+    }
+
+    final Point[] points = getRoutePoints(routeDescription, mapData);
```

Comment:
Avoid declaring all variables up front. This is not very cohesive. Cohesion keeps related entities close together, declaration + usage. As is, I have to search through the code base to see where these are used and to determine theri scope.
Easiest way to do this, cut the line out, see where you get the first compiler error, and insert the line above the line generating the error. Please do this for all the variables.